### DISCUSSION TITLE
What the current stage of V8?

### DISCUSSION THREAD

**Discussion by abouolia** [TanStack/table#3054] [General] [answered]
Actually this version has fundamental issue with controlled mode, i could not make any progress in our project cause depends heavily on outer state, just curious what the current stage of V8 to give some sponsors, help and testing to the next version.

**Comment by tannerlinsley** [comment, marked as answer]
It’s in an experimental phase, currently only visible to Bronze tier sponsors or higher.

**Reply by ledbetterljoshua** [reply]
If you published the new features V8 has along with a roadmap, people might be more willing to sponsor, depending on whether it has features they want or not.

**Reply by trumbitta** [reply]
I for one would advocate for my company to sponsor if I knew for sure that proper TS typings are coming soon with v8 as mentioned in some issues.

**Reply by tannerlinsley** [reply]
V8 will be fully typesafe and it is about 70% complete. I plan to ship hopefully Q1 of 2022

**Reply by marceloverdijk** [reply]
@tannerlinsley no pressure, but is Q1 2022 still the plan?

Would it be an idea to create milestones (e.g. beta, release) with a planned date?

**Reply by andreykovalenko** [reply]
@tannerlinsley would be awesome to get some feedback on the progress.

**Reply by osdiab** [reply]
Q1 has passed, Q2 is rapidly approaching - @tannerlinsley if any updated insight on timing that would be awesome :) my team is rearing to rewrite our not-type-safe tables in it but would love the assurance of it being official.

- A happy TanStack sponsor

**Reply by tannerlinsley** [reply]
I am tentatively planning a very-stable beta release (possibly even GC) mid-June. 🎉

**Reply by MAb540** [reply]
@tannerlinsley is v8 stable now??

**Reply by KevinVandy** [reply]
Has been since July
